### ISSUE TITLE
java.lang.ClassCastException: org.json.JSONObject$Null cannot be cast to java.util.ArrayList

### ISSUE THREAD

**Issue by bishnoirahul** [Azure/azure-cosmosdb-spark#261]
I am reading a collection with json object whose format is like below:
{
  "accountId" : "someguid",
  "emailId" : "a@abc.com",
  "status":"Success",
  "notes" :[
              { "Comment:"starting",
                "PostedBy":"user1", 
                "PostedOn" : "21 Dec 2017"}
    ]
}

i am reading this collection like below:
```
val applicationDocDBConnection = Config(Map("Endpoint" -> "https://somedb.documents.azure.com:443/",
"Masterkey" -> "somekey",
"Database" -> "somedatabase",
"PreferredRegions" -> "East US;",
"Collection" -> "somecollection",
"SamplingRatio" -> "1.0",
"schema_samplesize" -> "10",
  "query_custom" -> "SELECT c.accountId, c.status, c.notes, c.emailId FROM c" ))

val applications = spark.sqlContext.read.cosmosDB(applicationDocDBConnection);

applications.show()
```

If in collection any of the object does not has notes properties or when notes : [] , everything works fine.
But when in any object, **"notes" : null,** I get below exception when running cmd applications.show() - 

```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 14.0 failed 4 times, most recent failure: Lost task 0.3 in stage 14.0 (TID 29, 10.139.64.6, executor 0): java.lang.ClassCastException: org.json.JSONObject$Null cannot be cast to java.util.ArrayList
at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$toSQL$1.apply(CosmosDBRowConverter.scala:113)
	at scala.Option.map(Option.scala:146)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$.toSQL(CosmosDBRowConverter.scala:97)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1$$anonfun$apply$5.apply(CosmosDBRowConverter.scala:91)
	at scala.Option.map(Option.scala:146)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1.apply(CosmosDBRowConverter.scala:91)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1.apply(CosmosDBRowConverter.scala:82)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$.recordAsRow(CosmosDBRowConverter.scala:82)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$asRow$1.apply(CosmosDBRowConverter.scala:68)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$asRow$1.apply(CosmosDBRowConverter.scala:67)

```

Its a very basic use case when any array object can be null in document db. It should be fixed or is there any workaround?

**Comment by jpsimen** [comment]
I'm absolutely with you. Having the same issue. See #256

**Comment by tknandu** [comment]
@bishnoirahul @jpsimen Investigating this - will get back with resolution

**Comment by massoudm** [comment]
I have the same issue and it is not related to change feed. Is there a work around for non-streaming scenario?

**Comment by nomiero** [comment]
This should be fixed in the latest version 1.3.3. Can you please upgrade and try it ?

**Comment by PVikash** [comment]
I am using 1.3.4 version but facing a similar kind of issue.  
I am writing stream as 
```
myDF.writeStream
      .format("com.microsoft.azure.cosmosdb.spark.streaming.CosmosDBSinkProvider")
      .options(configMap)
      .option("checkpointLocation", checkpointPath + containerName + "/clean")
      .start()

```
if my input JSON data has a null value in an array like 
`"addresses" : [ null, null ] `
it is giving me following error:

> 19/03/15 10:51:12 INFO DAGScheduler: ResultStage 1 (start at LEDStreamJob.scala:110) failed in 5.105 s due to Job aborted due to stage failure: Task 0 in stage 1.0 failed 4 times, most recent failure: Lost task 0.3 in stage 1.0 (TID 5, wn1-ukw-de.4inaab3caa4uneflxy1g3a5fdd.cwx.internal.cloudapp.net, executor 2): java.lang.NullPointerException
>         at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$.com$microsoft$azure$cosmosdb$spark$schema$CosmosDBRowConverter$$convertToJson(CosmosDBRowConverter.scala:162)
> 

In batch processing(NOT streaming), If I use 
`df.write.mode(SaveMode.Overwrite).cosmosDB(writeConfig)` 
it is working fine for same data.
